plugin/plugin.go

```diff
@@ -68,22 +68,22 @@ func (DescribeKeyRequest) Command() Command {
 	return CommandDescribeKey
 }
 
-// GenerateSignatureResponse is the response of a describe-key request.
+// DescribeKeyResponse is the response of a describe-key request.
 type DescribeKeyResponse struct {
 	// The same key id as passed in the request.
 	KeyID string `json:"keyId"`
 
 	// One of following supported key types:
 	// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#algorithm-selection
-	KeySpec notation.KeySpec `json:"keySpec"`
+	KeySpec signer.KeySpec `json:"keySpec"`
 }
 
 // GenerateSignatureRequest contains the parameters passed in a generate-signature request.
 type GenerateSignatureRequest struct {
 	ContractVersion string                 `json:"contractVersion"`
 	KeyID           string                 `json:"keyId"`
-	KeySpec         notation.KeySpec       `json:"keySpec"`
-	Hash            notation.HashAlgorithm `json:"hashAlgorithm"`
```

Comment:
Should we keep well defined type of hash or an string works?
Decide to use string as we are internally using crypto.Hash to define type.